From 0ad96ee1f850275749de442089c003efda124044 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Mon, 13 Jul 2015 15:11:36 +0800
Subject: [PATCH] MLK-11240 video: mxsfb: refine the lcdif driver code logic

Do some code logic refinement for lcdif driver to make it
more reasonable and strong. set_par and enable_controller
calling sequences should be more exact.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/video/mxsfb.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/video/mxsfb.c b/drivers/video/mxsfb.c
index 43f6058..012e523 100644
--- a/drivers/video/mxsfb.c
+++ b/drivers/video/mxsfb.c
@@ -551,12 +551,6 @@ static void mxsfb_enable_controller(struct fb_info *fb_info)
 	}
 	clk_enable_pix(host);
 
-	/* Clean soft reset and clock gate bit if it was enabled  */
-	writel(CTRL_SFTRST | CTRL_CLKGATE, host->base + LCDC_CTRL + REG_CLR);
-
-	/* reconfigure the lcdif after */
-	mxsfb_set_par(&host->fb_info);
-
 	writel(CTRL2_OUTSTANDING_REQS__REQ_16,
 		host->base + LCDC_V4_CTRL2 + REG_SET);
 
@@ -872,6 +866,11 @@ static int mxsfb_blank(int blank, struct fb_info *fb_info)
 
 	case FB_BLANK_UNBLANK:
 		if (!host->enabled) {
+			clk_enable_pix(host);
+			clk_enable_axi(host);
+			clk_enable_disp_axi(host);
+
+			writel(0, host->base + LCDC_CTRL);
 			mxsfb_set_par(&host->fb_info);
 			mxsfb_enable_controller(fb_info);
 		}
-- 
1.8.0

